kitchen/modules/child-trie/src/lib.rs

```diff
@@ -1 +1,149 @@
-// child trie storage example
\ No newline at end of file
+/// Child Trie Storage Example (designed to be minimal and generic)
+/// - for efficient storage of (key, value) pairs when **proofs of key inclusion** need to be cheap
+///
+/// - the root of the child trie is used to prove participation of the `key`
+/// - the trie is retained for efficient usage of the associated `value` later
+use support::{
+	decl_module, decl_storage, decl_event, storage::child, ensure,
+	traits::Get, dispatch::Result,
+};
+use system::ensure_signed;
+use substrate_primitives::storage::well_known_keys::CHILD_STORAGE_KEY_PREFIX;
+use primitives::{Blake2Hasher, Hasher};
+
+// TODO: check if this is unnecessary dependency
+use parity_scale_codec::{Encode, Decode};
+use rstd::vec::Vec; // TODO: should consider replacing all `use rstd::prelude::*;` w/ this (consider a gist for this common question)?
+
+pub trait Trait: system::Trait {
+    type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+
+    type UpdateFrequency: Get<Self::BlockNumber>;
+}
+
+// this is the value type associated with the AccountId
+// is `BalanceOf<T>` in the parachain/runtime/crowdfund
+pub type ValueType = u32;
+pub type ExampleIndex = u32;
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub struct ExampleObject<AccountId, BlockNumber> {
+    /// initiated the example object which will accept new (key, value) submissions
+    initiator: AccountId,
+    /// sum_of_values (for example purposes)
+    sum: ValueType,
+    /// maximum sum allowed
+    max_sum: ValueType,
+    /// new submissions accepted until this block number
+    end: BlockNumber,
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait> as ChildTrie {
+        ExampleObjects get(example_objects):
+            map ExampleIndex => Option<ExampleObject<T::AccountId, T::BlockNumber>>;
+
+        ObjectCount get(object_count): ExampleIndex;
+
+        SpecialObjects get(special_objects): Vec<ExampleIndex>;
+    }
+}
+
+decl_event! {
+    pub enum Event<T> where
+	    <T as system::Trait>::AccountId,
+        <T as system::Trait>::BlockNumber,
+    {   
+        /// object initiated by AccountId
+        NewObject(ExampleIndex, AccountId),
+        /// (key, value) added to object
+        AppendVal(ExampleIndex, AccountId, ValueType),
```

Comment:
To be more aligned with substrate: use passive names here: `ValAppended`

(I hope this thing is called _passive_ in grammar. you get the point.)